var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -674,6 +674,12 @@ def configure_args(self):
         if spec.satisfies('+lustre'):
             lustre_opt = '--with-lustre={0}'.format(spec['lustre'].prefix)
             config_args.append(lustre_opt)
+            # Fixme: How to treat gpfs if it's on with lustre?
+            # Default is +gpfs, but development support may be absent.
+            if spec.satisfies('+gpfs'):
+                raise InstallError("Configuring ROMIO for both GPFS and Lustre not supported; use ~gpfs")
```

Comment:
Shouldn't this be a `conflicts('+gpfs', when='+lustre', msg='Configuring ROMIO...')`?

Reply:
> I don't see why lustre and gpfs should conflict, though that's an easy way out. I think we'll have both available on a system, although gpfs probably wouldn't be used for mpi-io.

@loveshack Your configuration script here (simplified) is:
```python
if spec.satisfies('+lustre'):
    if spec.satisfies('+gpfs'):
        raise InstallError("...")
```
Using a `conflicts` means that spack knows that the combination is invalid rather than having to install the whole spec up to this point to find out.